modules/core/src/copy.cpp

```diff
@@ -1328,7 +1328,7 @@ void copyMakeConstBorder_8u( const uchar* src, size_t srcstep, cv::Size srcroi,
     dst += dststep*top;
 
     for( i = 0; i < top; i++ )
-        memcpy(dst + (i - top)*dststep, constBuf, dstroi.width);
+        memcpy(dst + (i - top)*static_cast<int>(dststep), constBuf, dstroi.width);
 
     for( i = 0; i < bottom; i++ )
         memcpy(dst + (i + srcroi.height)*dststep, constBuf, dstroi.width);
```

Comment:
Please add problem description.

Are you sure that this patch fixes your problem? Because there is another similar code few lines below.

Reply:
I added a more detailed description of the problem I found.
I am sure it fixed the UB I found. This is the only part I am using so far in my code so I cannot tell whether there are/aren't more UB similar to this one, but at least this one is fixed. I will be more than happy to fix other as I cross them my way, though.

Reply:
I added a more detailed description of the problem.

I am sure it fixed the UB I found. This is the only part I am using so far in my code so I cannot tell whether there are/aren't more UB similar to this one, but at least this one is fixed. I will be more than happy to fix other as I cross them my way, though.

Reply:
The fix will fail if `dststep` is greater than `INT_MAX`. That's not the most often case, but possible.
Also `i-top` is always negative so the fix could be `dst - (top - i)*dststep`